adapters/tlsio_appleios.c

```diff
@@ -0,0 +1,508 @@
+#define CallOpenCallback(status) do { if (tlsio_instance->on_io_open_complete != NULL) (void)tlsio_instance->on_io_open_complete(tlsio_instance->on_io_open_complete_context, status); } while((void)0,0)
+#define CallCloseCallback() do { if (tlsio_instance->on_io_close_complete != NULL) (void)tlsio_instance->on_io_close_complete(tlsio_instance->on_io_close_complete_context); } while((void)0,0)
+
+
+typedef struct TLS_IO_INSTANCE_TAG
+{
+    ON_IO_OPEN_COMPLETE on_io_open_complete;
+    void* on_io_open_complete_context;
+
+    ON_BYTES_RECEIVED on_bytes_received;
+    void* on_bytes_received_context;
+
+    ON_IO_ERROR on_io_error;
+    void* on_io_error_context;
+
+    ON_IO_CLOSE_COMPLETE on_io_close_complete;
+    void* on_io_close_complete_context;
+
+    CFStringRef remote_host;
+    uint16_t port;
+	
+    CFReadStreamRef sockRead;
+    CFWriteStreamRef sockWrite;
+	
+    TLSIO_APPLEIOS_STATE state;
+} TLS_IO_INSTANCE;
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_008: [ The tlsio_appleios_get_interface_description shall return the VTable IO_INTERFACE_DESCRIPTION. ]*/
+const IO_INTERFACE_DESCRIPTION* tlsio_appleios_get_interface_description(void)
+{
+    return &tlsio_handle_interface_description;
+}
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_005: [ The tlsio_appleios shall received the connection information using the TLSIO_CONFIG structure defined in `tlsio.h`. ]*/
+/* Codes_SRS_TLSIO_APPLEIOS_32_009: [ The tlsio_appleios_create shall create a new instance of the tlsio for iOS. ]*/
+/* Codes_SRS_TLSIO_APPLEIOS_32_017: [ The tlsio_appleios_create shall receive the connection configuration (TLSIO_CONFIG). ]*/
+CONCRETE_IO_HANDLE tlsio_appleios_create(void* io_create_parameters)
+{
+    TLS_IO_INSTANCE* tlsio_instance;
+    if (io_create_parameters == NULL)
+    {
+        LogError("Invalid TLS parameters.");
+        tlsio_instance = NULL;
+    }
+    else
+    {
+        /* Codes_SRS_TLSIO_APPLEIOS_32_011: [ The tlsio_appleios_create shall allocate memory to control the tlsio instance. ]*/
+        tlsio_instance = (TLS_IO_INSTANCE*)malloc(sizeof(TLS_IO_INSTANCE));
+        if (tlsio_instance == NULL)
+        {
+            /* Codes_SRS_TLSIO_APPLEIOS_32_012: [ If there is not enough memory to control the tlsio, the tlsio_appleios_create shall return NULL as the handle. ]*/
+            LogError("There is not enough memory to create the TLS instance.");
+            /* return as is */
+        }
+        else
+        {
+            /* Codes_SRS_TLSIO_APPLEIOS_32_005: [ The tlsio_appleios shall received the connection information using the TLSIO_CONFIG structure defined in `tlsio.h`. ]*/
+            /* Codes_SRS_TLSIO_APPLEIOS_32_017: [ The tlsio_appleios_create shall receive the connection configuration (TLSIO_CONFIG). ]*/
+            TLSIO_CONFIG* tlsio_config = (TLSIO_CONFIG*)io_create_parameters;
+            
+            if (tlsio_config->hostname == NULL)
+            {
+                LogError("Host name was not provided");
```

Comment:
>LogError [](start = 16, length = 8)

You need to take care of this error. Please free the tlsio_instance memory and make it equals NULL.